speech/check_dataVals.m

```diff
@@ -414,51 +487,48 @@ function update_plots(src,evt)
+        set(UserData.htracks(iPlot).(UserData.sigs2plot{s})(selF==src),'Color',plotcolor{s},'LineWidth',3)
+        handleName = strcat(UserData.sigs2plot{s},'Ends');
+        set(UserData.htracks(iPlot).(handleName)(selF==src),'MarkerEdgeColor',plotcolor{s},'MarkerFaceColor',get_lightcolor(plotcolor{s},1.2))
+        
+        uistack(UserData.htracks(iPlot).(UserData.sigs2plot{s})(selF==src),'top');
+        uistack(UserData.htracks(iPlot).(handleName)(selF==src),'top');
+        
+        set(UserData.htracks(iPlot).(UserData.sigs2plot{s})(selF~=src),'Color',unselectedColor,'LineWidth',1)
+        set(UserData.htracks(iPlot).(handleName)(selF~=src),'MarkerEdgeColor',unselectedColor,'MarkerFaceColor',get_lightcolor(unselectedColor,1.2))
     end
-    set(UserData.htracks(iPlot).f1(selF==src),'Color',f1color,'LineWidth',3)
-    set(UserData.htracks(iPlot).f2(selF==src),'Color',f2color,'LineWidth',3)
-    set(UserData.htracks(iPlot).f1Ends(selF==src),'MarkerEdgeColor',f1color,'MarkerFaceColor',get_lightcolor(f1color,1.2))
-    set(UserData.htracks(iPlot).f2Ends(selF==src),'MarkerEdgeColor',f2color,'MarkerFaceColor',get_lightcolor(f2color,1.2))
-    uistack(UserData.htracks(iPlot).f1(selF==src),'top');
-    uistack(UserData.htracks(iPlot).f2(selF==src),'top');
-    uistack(UserData.htracks(iPlot).f1Ends(selF==src),'top');
-    uistack(UserData.htracks(iPlot).f2Ends(selF==src),'top');
-    
-    set(UserData.htracks(iPlot).f1(selF~=src),'Color',unselectedColor,'LineWidth',1)
-    set(UserData.htracks(iPlot).f2(selF~=src),'Color',unselectedColor,'LineWidth',1)
-    set(UserData.htracks(iPlot).f1Ends(selF~=src),'MarkerEdgeColor',unselectedColor,'MarkerFaceColor',get_lightcolor(unselectedColor,1.2))
-    set(UserData.htracks(iPlot).f2Ends(selF~=src),'MarkerEdgeColor',unselectedColor,'MarkerFaceColor',get_lightcolor(unselectedColor,1.2))
+
     for i = 1:length(UserData.htracks)
         if i ~= iPlot
-            set(UserData.htracks(i).f1(:),'Color',unselectedColor,'LineWidth',1)
-            set(UserData.htracks(i).f2(:),'Color',unselectedColor,'LineWidth',1)
-            set(UserData.htracks(i).f1Ends(:),'MarkerEdgeColor',unselectedColor,'MarkerFaceColor',get_lightcolor(unselectedColor,1.2))
-            set(UserData.htracks(i).f2Ends(:),'MarkerEdgeColor',unselectedColor,'MarkerFaceColor',get_lightcolor(unselectedColor,1.2))
+            for s = 1:UserData.nSigs
+                handleName = strcat(UserData.sigs2plot{s},'Ends');
+                set(UserData.htracks(i).(UserData.sigs2plot{s})(:),'Color',unselectedColor,'LineWidth',1)
+                set(UserData.htracks(i).(handleName)(:),'MarkerEdgeColor',unselectedColor,'MarkerFaceColor',get_lightcolor(unselectedColor,1.2))
+            end
         end
     end
     guidata(src,UserData);
 end
 
 function TB_all(src,evt)
     UserData = guidata(src);
-    f1color = [0 0 1]; % blue
-    f2color = [1 0 0]; % red
+    plotcolor{1} = [0 0 1]; % blue
+    plotcolor{2} = [1 0 0]; % red
```

Comment:
Also here I guess

Reply:
Added 3rd plot color in c2f23bf3d389ad07c2f5576d28dc3eb2257793da